syntax = "proto3";

package suggestions;

service SuggestionsService {
    rpc SuggestBooks (SuggestionRequest) returns (SuggestionResponse);
}

message SuggestionRequest {
    string card_number = 1;
    float order_amount = 2;
}

message Book {
    string id = 1;
    string author = 2;
    string title = 3;
}

message SuggestionResponse {
    repeated Book books = 1;
}


service HelloService {
    rpc SayHello (HelloRequest) returns (HelloResponse);
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string greeting = 1;
}